src/metabase/driver/sql/query_processor.clj

```diff
@@ -1411,6 +1423,17 @@
   ;; athena cannot cast uuid to bounded varchars
   (->honeysql driver [::cast expr "text"]))
 
+(defmethod ->honeysql [:sql ::expression-literal-text-value]
```

Comment:
There are multiple ways to slice this, and I'm open to other ideas. This implementation is following the example of `::cast-to-text` just above. Another option is to add a new multimethod like `literal-string-expression->honeysql`, similar in spirit to `unix-timestamp->honeysql`. You could also imagine making this more generic and calling it on *any* literal expression, not just strings. In practice, only strings seem to cause problems since other literals get inlined directly, but in that case it would be up to the driver to check the type of the `:value` and decide if it needs special handling for, e.g., boolean literals.